lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2091,21 +2097,27 @@ def draw_path_collection(self, gc, master_transform, paths, all_transforms,
         #    uses_per_path for the uses
         len_path = len(paths[0].vertices) if len(paths) > 0 else 0
         uses_per_path = self._iter_collection_uses_per_path(
-            paths, all_transforms, offsets, facecolors, edgecolors)
+            paths, all_transforms, offsets, vgc.get_facecolors(), vgc.get_edgecolors())
         should_do_optimization = \
             len_path + uses_per_path + 5 < len_path * uses_per_path
 
         if (not can_do_optimization) or (not should_do_optimization):
             return RendererBase.draw_path_collection(
-                self, gc, master_transform, paths, all_transforms,
-                offsets, offset_trans, facecolors, edgecolors,
-                linewidths, linestyles, antialiaseds, urls,
-                offset_position, hatchcolors=hatchcolors)
+                self, vgc, master_transform, paths, all_transforms,
+                offsets, offset_trans)
+
+        Njoinstyles = len(vgc._joinstyles)
+        Ncapstyles = len(vgc._capstyles)
 
-        padding = np.max(linewidths)
+        padding = np.max(vgc.get_linewidths())
         path_codes = []
         for i, (path, transform) in enumerate(self._iter_collection_raw_paths(
                 master_transform, paths, all_transforms)):
+            gc = self.new_gc()
+            if Njoinstyles:
```

Comment:
The creation of the new `gc` object should be done outside the loop.
```suggestion
        path_codes = []
        gc = self.new_gc()
        for i, (path, transform) in enumerate(self._iter_collection_raw_paths(
                master_transform, paths, all_transforms)):
            if Njoinstyles:
```